VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/General/ElementaryMethodsOfTypeInCollectionNotOverriddenTests.cs

```diff
@@ -0,0 +1,314 @@
+        }
+
+        [TestMethod]
+        public void ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer_WithReferenceType_ImplementsMethods()
+        {
+            var original = @"
+using System.Collections.Generic;
+namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        void Method()
+        {
+            var list = new List<MyCollectionItem>();
+        }
+    }
+
+    class MyCollectionItem
+    {
+        public override bool Equals(object obj)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override int GetHashCode()
+        {
+            throw new System.NotImplementedException();
+        }
+    }
+}";
+
+            VerifyDiagnostic(original);
+        }
+
+        [TestMethod]
+        public void ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer_WithValueType_ImplementsMethods()
+        {
+            var original = @"
+using System.Collections.Generic;
+namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        void Method()
+        {
+            var list = new List<MyCollectionItem>();
+        }
+    }
+
+    struct MyCollectionItem
+    {
+        public override bool Equals(object obj)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public override int GetHashCode()
+        {
+            throw new System.NotImplementedException();
+        }
+    }
+}";
+
+            VerifyDiagnostic(original);
+        }
+
+        [TestMethod]
+        public void ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer_Dictionary_BothDoNotImplementMethods()
+        {
+            var original = @"
+using System.Collections.Generic;
+namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        void Method()
+        {
+            var list = new Dictionary<MyCollectionItem, MyCollectionItem>();
+        }
+    }
+
+    class MyCollectionItem {}
+}";
+
+            VerifyDiagnostic(original,
+                ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer.Rule.MessageFormat.ToString(),
+                ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer.Rule.MessageFormat.ToString());
+        }
+
+        [TestMethod]
+        public void ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer_Dictionary_OneDoesNotImplementMethods()
+        {
+            var original = @"
+using System.Collections.Generic;
+namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        void Method()
+        {
+            var list = new Dictionary<int, MyCollectionItem>();
+        }
+    }
+
+    class MyCollectionItem {}
+}";
+
+            VerifyDiagnostic(original,
+                ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer.Rule.MessageFormat.ToString());
+        }
+
+        [TestMethod]
+        public void ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer_IEnumerable()
```

Comment:
I would reword the test name to `ElementaryMethodsOfTypeInCollectionNotOverriddenAnalyzer_TypeParameterWithoutObjectCreation`